### PR TITLE
Gh 2555/bf/fix training log file

### PR DESCRIPTION
**Pull request by helpmefindaname**
As https://github.com/flairNLP/flair/issues/2555 is a bug introduced by https://github.com/flairNLP/flair/pull/2537 I created a fix and added a test to ensure that mistake won't be done again

### CODE REVIEW

**Comment by alanakbik** [comment]
Thanks for this fix @helpmefindaname! @whoisjones can you check this PR?

File: flair/models/diagnosis/distance_prediction_model.py [hunk 1]
```diff
@@ -464,7 +464,7 @@ def evaluate_classification(
                 log_line=log_line,
                 log_header=log_header,
                 detailed_results=detailed_result,
-                loss=eval_loss,
+                loss=eval_loss.item(),
```
Comment (whoisjones) [inline flair/models/diagnosis/distance_prediction_model.py:467]: looks good

File: flair/nn/model.py [hunk 2]
```diff
@@ -390,7 +390,7 @@ def evaluate(
             log_header=log_header,
             detailed_results=detailed_result,
             classification_report=classification_report_dict,
-            loss=eval_loss,
+            loss=eval_loss.item(),
```
Comment (whoisjones) [inline flair/nn/model.py:393]: looks good

File: flair/trainers/trainer.py [hunk 3]
```diff
@@ -385,6 +363,27 @@ def train(
             else:
                 log_handler = None
 
+            log_line(log)
```
Comment (whoisjones) [inline flair/trainers/trainer.py:366]: 👍

File: tests/test_trainer.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import pytest
```
Comment (whoisjones) [inline tests/test_trainer.py:1]: 👍

**Comment by whoisjones** [comment]
👍

**Comment by alanakbik** [comment]
:+1:

**Comment by helpmefindaname** [comment]
You are welcome, and if you find other bugs, that are likely introduced by me just feel free to ping me

**Pull request merged**